code/game/machinery/turrets.dm

```diff
@@ -336,157 +362,122 @@
 	else if(ispath(control_area))
 		control_area = locate(control_area)
 
-	return
+	updateTurrets() //Updates the turrets and the icon if an instance is made that is not set to stun by default
 
 /obj/machinery/turretid/emag(mob/user)
 	if(!emagged)
 		emagged = 1
 		locked = 0
 		if(user)
-			to_chat(user, "<span class='warning'>You short out the turret controls' access analysis module.</span>")
-			if(user.machine==src)
-				src.attack_hand(user)
+			to_chat(user, "<span class='warning'>You short out [src]'s access analysis and threat indicator module.</span>")
+			if(user.machine == src)
+				attack_hand(user)
+			update_icons() //Update the icon immediately since emagging removes the turret threat indicator
 		return 1
 	return
 
 /obj/machinery/turretid/attackby(obj/item/weapon/W, mob/user)
 	if(stat & BROKEN)
 		return
-	if (istype(user, /mob/living/silicon))
-		return src.attack_hand(user)
+	if(issilicon(user))
+		return attack_hand(user)
 
 	..()
 
-	if( get_dist(src, user) == 0 )		// trying to unlock the interface
-		if (src.allowed(usr))
+	if(user.Adjacent(src))
+		if(allowed(user))
 			if(emagged)
-				to_chat(user, "<span class='notice'>The turret control is unresponsive.</span>")
+				to_chat(user, "<span class='warning'>[src]'s card reader is totally unresponsive.</span>")
 				return
 
 			locked = !locked
-			to_chat(user, "<span class='notice'>You [ locked ? "lock" : "unlock"] the panel.</span>")
-			if (locked)
-				if (user.machine==src)
+			to_chat(user, "<span class='notice'>You [locked ? "lock" : "unlock"] the switchboard panel.</span>")
+			if(locked)
+				if (user.machine == src)
 					user.unset_machine()
 					user << browse(null, "window=turretid")
 			else
-				if (user.machine==src)
+				if(user.machine == src)
 					src.attack_hand(user)
 		else
 			to_chat(user, "<span class='warning'>Access denied.</span>")
 
 /obj/machinery/turretid/attack_ai(mob/user as mob)
-	src.add_hiddenprint(user)
+	add_hiddenprint(user)
 	if(!ailock || isAdminGhost(user))
 		return attack_hand(user)
 	else
-		to_chat(user, "<span class='notice'>There seems to be a firewall preventing you from accessing this device.</span>")
+		to_chat(user, "<span class='notice'>There seems to be a firewall preventing you from accessing [src].</span>")
 
 /obj/machinery/turretid/attack_hand(mob/user as mob)
-	if ( get_dist(src, user) > 0 )
-		if ( !issilicon(user) && !isAdminGhost(user))
+	if(!user.Adjacent(src))
+		if(!issilicon(user) && !isAdminGhost(user))
 			to_chat(user, "<span class='notice'>You are too far away.</span>")
 			user.unset_machine()
 			user << browse(null, "window=turretid")
 			return
 
 	user.set_machine(src)
+	// Oh god what
 	var/loc = src.loc
-	if (istype(loc, /turf))
+	if(istype(loc, /turf))
 		loc = loc:loc
-	if (!istype(loc, /area))
-		to_chat(user, text("Turret control badly positioned - loc.loc is [].", loc))
+	if(!istype(loc, /area))
+		to_chat(user, "Turret control badly positioned - loc.loc is [loc].") //Debug message
```

Comment:
Make it something a bit more IC. `Turret control must be placed on a tile.`
Also, I _think_ this whole block simply checks if the controller is placed on a turf and if that turf isn't in space